projects/packages/videopress/src/client/state/reducers.js

```diff
@@ -288,6 +281,84 @@ const videos = ( state, action ) => {
 			};
 		}
 
+		case UPLOADING_VIDEO: {
+			const { id, title } = action;
+			const currentMeta = state?._meta || {};
+			const currentMetaItems = currentMeta?.items || {};
+
+			return {
+				...state,
+				_meta: {
+					...currentMeta,
+					items: {
+						...currentMetaItems,
+						[ id ]: {
+							title,
+							uploading: true,
+						},
+					},
+				},
+			};
+		}
+
+		case PROCESSING_VIDEO: {
+			const { id, data } = action;
+			const items = [ ...( state?.items ?? [] ) ];
+			const currentMeta = state?._meta || {};
+			const currentMetaItems = Object.assign( {}, currentMeta?.items || {} );
+			const title = currentMetaItems[ id ]?.title || '';
+
+			// Insert new video
+			items.unshift( {
+				id: data.id,
+				guid: data.guid,
+				url: data.src,
+				title,
+				posterImage: null,
+				finished: false,
+			} );
+
+			// Remove video from uploading meta
+			delete currentMetaItems[ id ];
+
+			return {
+				...state,
+				items,
+				_meta: {
+					...currentMeta,
+					items: currentMetaItems,
+				},
+			};
```

Comment:
Could we track the processing data too?
```suggestion
			// Remove video from uploading meta
			delete currentMetaItems[ id ];

			return {
				...state,
				items,
				_meta: {
					...currentMeta,
					items: {
						...currentMetaItems,
						[ data.id ]: {
							hasUploaded: true,
							isProcessing: false,
						},
					},
				},
			};
```

https://user-images.githubusercontent.com/77539/194558348-f6942937-71ab-4e79-85c6-7ed370ede52b.mov